p2p/conn/connection.go

```diff
@@ -779,6 +779,10 @@ type Channel struct {
 	sending       []byte
 	recentlySent  int64 // exponential moving average
 
+	nextPacketMsg           *tmp2p.PacketMsg
```

Comment:
Can you add a comment above mentioning that we reuse the same packets for sending purposes?

Reply:
Moreover, why we need pointers here? We allocate these fields only once, right?

Reply:
We need the pointer since we need to store the on heap copy. (If we stored the stack object, then if we're not careful we may take a copy to stack and then push that to heap on every pointer call)

We could just be careful but seems easier to use a pointer and not have to think about it